pr2_moveit_tutorials/pick_place/src/pick_place_tutorial.cpp

```diff
@@ -43,13 +43,52 @@
 
 static const std::string ROBOT_DESCRIPTION="robot_description";
 
+void openGripper(trajectory_msgs::JointTrajectory& posture){
+  posture.joint_names.resize(6);
+  posture.joint_names[0] = "r_gripper_joint";
+  posture.joint_names[1] = "r_gripper_motor_screw_joint";
+  posture.joint_names[2] = "r_gripper_l_finger_joint";
+  posture.joint_names[3] = "r_gripper_r_finger_joint";
+  posture.joint_names[4] = "r_gripper_r_finger_tip_joint";
+  posture.joint_names[5] = "r_gripper_l_finger_tip_joint";
+
+  posture.points.resize(1);
+  posture.points[0].positions.resize(6);
+  posture.points[0].positions[0] = 1;
+  posture.points[0].positions[1] = 1.0;
+  posture.points[0].positions[2] = 0.477;
+  posture.points[0].positions[3] = 0.477;
+  posture.points[0].positions[4] = 0.477;
+  posture.points[0].positions[5] = 0.477;
+}
+
+void closedGripper(trajectory_msgs::JointTrajectory& posture){
+  posture.joint_names.resize(6);
+  posture.joint_names[0] = "r_gripper_joint";
+  posture.joint_names[1] = "r_gripper_motor_screw_joint";
+  posture.joint_names[2] = "r_gripper_l_finger_joint";
+  posture.joint_names[3] = "r_gripper_r_finger_joint";
+  posture.joint_names[4] = "r_gripper_r_finger_tip_joint";
+  posture.joint_names[5] = "r_gripper_l_finger_tip_joint";
+
+  posture.points.resize(1);
+  posture.points[0].positions.resize(6);
+  posture.points[0].positions[0] = 0;
+  posture.points[0].positions[1] = 0;
```

Comment:
On Thu, Jul 28, 2016 at 07:39:42AM -0700, Christian Dornhege wrote:
> Where did you get these values from? I am using 0, 0.002, 0.002, 0.002, 0.002. I believe these were determine from an actual closing in gazebo.

I guessed :-)

You are short by one number. The gripper has 6 joints, not 5.
If you could give me a full list of joint-values for a closed state, I'm happy to amend the request.